include/deal.II/grid/grid_tools.h

```diff
@@ -756,6 +756,86 @@ namespace GridTools
   std::vector<typename MeshType::active_cell_iterator>
   compute_ghost_cell_halo_layer (const MeshType &mesh);
 
+  /**
+   * Extract and return the active cell within a @p layer_thickness
+   * around a subdomain (set of active cells) in the @p mesh.
+   * Here, the "subdomain" consists of exactly all of
+   * those cells for which the @p predicate returns @p true.
+   *
+   * The function first computes interior active, boundary cells of the
+   * subdomain that conform to the predicate.
+   * Using compute_bounding_box(), a bounding box is
+   * computed for this subdomain and extended by @p layer_thickness. All the
+   * active cells that are outside the extended bounding box are ignored.
```

Comment:
If a cell is seen as a closed set, by outside I mean that the intersection of the cell with the extended bounding box is a null set. It is inside otherwise. I understand that this terminology isn't general.  Numerically I am ensuring this by doing the following: 
```
for (unsigned int d = 0; d < spacedim; ++d)
  cell_inside &= (cell->center()[d] + cell_radius > bounding_box.first[d])
                    && (cell->center()[d] - cell_radius < bounding_box.second[d]);
// cell_inside is true if any of the cell vertices are inside the extended bounding box
```
which is not exactly checking if each vertex is inside. I suppose it's a bit more conservative than that. One could also check all vertices. But I thought may be this way is faster. I am open to change it the way dealii members suggest.